adafruit_ht16k33/segments.py

```diff
@@ -343,6 +350,9 @@ def _put(self, char: str, index: int = 0):
         # pylint: disable=too-many-return-statements
         if not 0 <= index <= 3:
             return
+        if self._chardict and char in self._chardict:
+            self._set_buffer(index, self._chardict[char])
```

Comment:
I dug for a bit and found that it is making it to here and calling `_set_buffer()` with values that seem okay to me, but nothing seems to happen on the segment display. Possibly it's an auto_write issue?

I'm putting it down for now but will circle back eventually to look more if you don't end up getting a solution once you've got your hardware for testing.